module/VuFind/src/VuFind/Controller/FeedbackController.php

```diff
@@ -96,49 +98,20 @@ public function formAction()
             return $view;
         }
 
-        $fields = $form->mapRequestParamsToFieldValues($this->params()->fromPost());
-        $emailMessage = $this->getViewRenderer()->partial(
-            'Email/form.phtml',
-            compact('fields')
-        );
-
-        [$senderName, $senderEmail] = $this->getSender($form);
-
-        $replyToName = $params->fromPost(
-            'name',
-            $user ? trim($user->firstname . ' ' . $user->lastname) : null
-        );
-        $replyToEmail = $params->fromPost(
-            'email',
-            $user ? $user->email : null
-        );
-
-        $recipients = $form->getRecipient($params->fromPost());
-
-        $emailSubject = $form->getEmailSubject($params->fromPost());
-
-        $sendSuccess = true;
-        foreach ($recipients as $recipient) {
-            [$success, $errorMsg] = $this->sendEmail(
-                $recipient['name'],
-                $recipient['email'],
-                $senderName,
-                $senderEmail,
-                $replyToName,
-                $replyToEmail,
-                $emailSubject,
-                $emailMessage
+        $primaryHandler = $form->getPrimaryHandler();
+        $success = $primaryHandler->handle($form, $params, $user ?: null);
+        if ($success) {
+            $view->setVariable('successMessage', $form->getSubmitResponse());
+            $view->setTemplate('feedback/response');
+        } else {
+            $this->flashMessenger()->addErrorMessage(
+                $this->translate('could_not_process_feedback')
             );
-
-            $sendSuccess = $sendSuccess && $success;
-            if (!$success) {
-                $this->flashMessenger()->addErrorMessage($errorMsg);
-            }
         }
 
-        if ($sendSuccess) {
-            $view->setTemplate('feedback/response');
-            $view->emailMessage = $emailMessage;
+        $handlers = $form->getSecondaryHandlers();
+        foreach ($handlers as $handler) {
+            $handler->handle($form, $params, $user ?: null);
```

Comment:
Might we want to put the secondary handling in a try..catch so that a fatal error in a secondary handler doesn't break the whole process?

Reply:
Done